### PR TITLE
Auth Refactor

### PR DESCRIPTION
**Pull request by minocys**
#### Auth Package
1. Move JWT services into auth folder
1. Refactor cookies generation into MakeCookies
1. Create authError object to hold error status
1. Add CheckOwnership, CheckMembership, checkProtected functions
1. Delete JWTUser type, use User model instead
1. Fetch array of owned instances when logging in
1. Add Owned and Members fields to user struct

#### ROUTES
1. Replace shortCode param in most routes with instanceID
1. Add JoinInstance route (no use case at the moment - instances are not protected)
1. Add instanceID to image routes - mostly not used at the moment but necessary for future authentication if implemented
1. Refactor instance controller auth logic into CheckOwnership

#### DATABASE
1. Move database creation into init.sql - had some issues initializing database with flyway
1. Add password field to instance

#### MISC
1. Add utils folder
1. Add DecodeBody util function
1. Add GetUserData for parsing Request context
1. Add CreateUserContext for adding user claims from jwt to request context

### CODE REVIEW

**Comment by minocys** [comment]
Not mergable atm, will need to remove further references to auth.JWTUser

**Comment by minocys** [comment]
rebased

File: auth/checks.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		// Proceed if not protected
+		if !cp {
+			next(w, r, p)
+			return
+		}
+
+		u, err := validateTokens(r)
+
+		if err != nil {
+			// Check if route is protected
```
Comment (hibearpanda) [inline auth/checks.go]: This comment does not belong here

File: router/routes.go [hunk 2]
```diff
@@ -26,47 +26,49 @@ func InitRouter(db *sql.DB, tc, hc *storage.Container) *httprouter.Router {
...
 	// Instance Routes
 	r.GET("/check/instances", ic.CheckShortCodeExistsHandler)
 	r.GET("/shortCode/:id", ic.FetchShortCodeByIDHandler)
 	r.GET("/instances", ic.GetInstancesHandler)
-	r.GET("/instances/:shortCode", ic.GetInstanceHandler)
-	r.PUT("/instances/:shortCode/title", ac.CheckRole(1, ic.UpdateInstanceTitleHandler))
-	r.PUT("/instances/:shortCode/font", ac.CheckRole(1, ic.UpdateInstanceFontFamilyHandler))
-	r.POST("/instances/:shortCode/hero", ac.CheckRole(1, ic.UpdateInstanceHeroHandler))
-	r.DELETE("/instances/:shortCode/hero", ac.CheckRole(1, ic.RemoveInstanceHeroHandler))
+	// Was instances/:shortCode
```
Comment (hibearpanda) [inline router/routes.go]: Just remove this comment

**Pull request merged**